### ISSUE TITLE
Bump pandas dependency

### ISSUE THREAD

**Issue by esalehim** [bp/ResSimpy#288]
Is there a reason for `ressimpy` being locked to pandas v1? Would be nice to loosen up the pandas dependency if possible, so that people can upgrade to pandas v2.

**Describe the solution you'd like**
`pandas = ">=1.5,<3.0"`

**Comment by jonnyh404** [comment]
I think the core reason is one of our dependencies (resqpy) requires a pandas 1.5 and would break if we were to move to pandas v2. 

Users are able to upgrade to pandas v2 post `poetry install`. But we don't guarantee that the resqpy function calls will then work.

**Comment by esalehim** [comment]
Hi @jonnyh404 , I see that `resqpy` is compatible with pandas v2 now. Could we now remove the pandas v1 lock here as well?

**Comment by jonnyh404** [comment]
Yes I think we should. There are some breaking changes to pandas from v1 to v2 that we'll have to work through.

**Comment by jonnyh404** [comment]
Completed with #430
